### ISSUE TITLE
Locally run a STAC API implementation and load our catalog into the backend

### ISSUE THREAD

**Issue by jtherrmann** [ASFHyP3/asf-stac#1]
We cloned <https://github.com/stac-utils/stac-fastapi> and ran the following command, which successfully starts two containers, `stac-fastapi` (for the API) and `pgstac` (for the Postgres database):

```
docker compose up app-pgstac
```

We were able to connect to the database with the following command (see the [`database` service](https://github.com/stac-utils/stac-fastapi/blob/master/docker-compose.yml#L64)):

```
PGUSER=username PGPASSWORD=password PGHOST=localhost PGDATABASE=postgis PGPORT=5439 psql
```

We observed that the database contains three tables: `geography_columns`, `geometry_columns`, and `spatial_ref_sys`.

We also confirmed that the `pgstac` container stores the database in a Docker volume, so that it persists across container restarts.

We're currently attempting to load the demo dataset into our database using the [`loadjoplin-pgstac` service](https://github.com/stac-utils/stac-fastapi/blob/master/docker-compose.yml#L98).

TODO:

- [x] Load the demo dataset into our database.
- [x] Query our STAC API endpoints against the demo data (see <https://github.com/radiantearth/stac-api-spec>)
- [x] Determine how to load our catalog into the database via API endpoints.
- [x] Determine why the `spatial_ref_sys` table contains `8500` rows by default, and whether that's some sort of demo data or if it's actually required for the STAC API to function.
- [x] Determine where the STAC data gets stored when you input it via the API (as the demo ingest script does). We're not seeing obvious changes to the database (`spatial_ref_sys` table still shows `8500` rows before and after the ingest) but it could be stored elsewhere in the Docker volume (because it appears to persist across container restarts)? I'm really curious where the data is actually stored and if it's not in the database, then what is the purpose of the database?
- [x] Merge https://github.com/ASFHyP3/asf-stac/pull/3

**Comment by forrestfwilliams** [comment]
[This](https://github.com/radiantearth/stac-api-spec/tree/main/item-search) looks like information on how to perform a search against a running api.

**Comment by forrestfwilliams** [comment]
It looks like creating a version of [ingest_joplin.py](https://github.com/stac-utils/stac-fastapi/blob/master/scripts/ingest_joplin.py) might be the way to go.

**Comment by forrestfwilliams** [comment]
Working off of [this stac-fastapi branch](https://github.com/forrestfwilliams/stac-fastapi/tree/get_it_working). Still need to modify `post_put` function.

**Comment by jtherrmann** [comment]
These sections of the API spec may be relevant to the ingest script (see https://github.com/ASFHyP3/asf-stac/pull/3) and its assumption that the STAC dataset is a tree:

* https://github.com/radiantearth/stac-api-spec/tree/v1.0.0-rc.1/core#structuring-catalog-hierarchies
* https://github.com/radiantearth/stac-api-spec/blob/v1.0.0-rc.1/browseable/README.md

**Comment by jtherrmann** [comment]
Investigating where the STAC data gets stored in the database:
* I confirmed that the `spatial_ref_sys` table, which contains 8500 rows, is unchanged after adding our STAC dataset. Upon further inspection, it looks like it just contains information on various spatial reference systems (hence the name of the table).
* I realized that `geometry_columns` (which is a [view](https://hasura.io/docs/latest/guides/postgres/views/), not a table) gains one row per STAC collection added via the API, so at least that confirms that the API is making changes to the database when we add our data (though I'm still not sure how the contents of `geometry_columns` actually correspond to our data).

I think I'm satisfied that the `pgstac` backend is working properly, though I don't understand how it works.
